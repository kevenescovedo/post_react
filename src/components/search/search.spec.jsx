import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { SearchComponent } from "."
const props = {
    posts: [{
        id: 1,
        title: 'title',
        body : 'body',
        cover: 'https://site.com',
    
    },
    {
        id: 2,
        title: 'title',
        body : 'body',
        cover: 'https://site2.com',
    
    }, 
    {
        id: 3,
        title: 'title',
        body : 'body',
        cover: 'https://site3.com',
    
    }
]
}
describe('</SearchComponent>', () => {
    it('should have a value of search value', () => {
        const fn = jest.fn();
      render(<SearchComponent handleChange={fn} searchValue="testando"/>);
        expect(screen.getByPlaceholderText("digite o que pesquisar")).toBeInTheDocument();
       // expect(screen.getByPlaceholderText("digite o que pesquisar")).toHaveAttribute('value','testando');
       expect(screen.getByPlaceholderText("digite o que pesquisar").value).toBe("testando");
    })
    
    it('should call handleChnage on each key press', () => {
        const fn = jest.fn();
       
      render(<SearchComponent handleChange={fn} />);
      const input = screen.getByPlaceholderText("digite o que pesquisar");
      userEvent.type(input,'o valor');
    let valor = 'o valor';
   // eu espero que o valor seja como esperado depois de digitar
   expect(input.value).toBe('o valor');
   //ve quantas vezes a função echmada
   expect(fn).toHaveBeenCalledTimes(valor.length);
    })
    it('should match snapshot', () => {
        const fn = jest.fn();
       
      const {debug, container}= render(<SearchComponent handleChange={fn} />);
      expect(container).toMatchSnapshot();
   
    })


   
});