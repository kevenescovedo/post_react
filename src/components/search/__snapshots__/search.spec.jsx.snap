// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</SearchComponent> should match snapshot 1`] = `
<div>
  <div
    class="caixa_search"
  >
    <div>
      hhhhhhh
    </div>
    <input
      class="text_input"
      placeholder="digite o que pesquisar"
      type="search"
      value=""
    />
     
    <br />
     
    <br />
     
    <br />
  </div>
</div>
`;
